src/initializers/kafka/base-kafka-handler.ts

```diff
@@ -72,24 +73,28 @@ abstract class Base<Input, Output> {
     return buffer;
   }
 
-  parseHeaders(headers: KafkajsType.IHeaders) {
-    return Object.keys(headers).reduce((m, k) => ({ ...m, [k]: headers[k].toString() }), {});
+  parseHeaders(headers: KafkajsType.IHeaders = {}) {
+    return Object.keys(headers).reduce((m, k) => {
+      m[k] = headers[k].toString();
+      return m;
+    }, {});
   }
+
   transformToKafkaHandlerMessage(
     message: KafkajsType.KafkaMessage,
     topic: string,
     partition: number
   ): BaseKafkaHandlerMessage<Input> {
     return {
       ...message,
+      key: message.key.toString(),
       topic,
       partition,
       ...(this.jsonParseValue && message.value
         ? { rawValue: message.value, value: this.parseValue(message.value) }
         : {}),
-      ...(this.stringifyHeaders && message.headers
-        ? { rawHeaders: message.headers, headers: this.parseHeaders(message.headers) }
-        : {})
+      rawHeaders: message.headers,
+      headers: this.parseHeaders(message.headers)
```

Comment:
As you removed condition for headers, do we need the same for rawValue above?

Reply:
Headers are buffers and toString() should always safe.
Kafka value could not be a json so parsing it to json would not be needed if the value someone uses is not json.
So no there is no need to remove it for value, as for binary for example content it would be an overkill to try json parse the messages.